### DISCUSSION TITLE
how to connect to celery?

### DISCUSSION THREAD

**Discussion by encryptblockr** [sabuhish/fastapi-mail#64] [General]
I will like to handle sending the emails to celery tasks
issue is when i do that, it complains that the sending email never awaited

how can one use this for non-async use case like celery?

**Comment by sabuhish** [comment]
I am not familiar with using non blocking tasks within celery, not totaly sure.

**Reply by maestro-1** [reply]
@sabuhish  Celery does not support async tasks that I am aware of. This is one of the benefits of splitting the connection the way I did [here](https://github.com/sabuhish/fastapi-mail/pull/122). Now customizable connection objects can be used. And a single function can be used included in the fastapi-mail class to allow calling sync functions. This will allow use with celery, background tasks and normal async.

It might be worth having a longer discussion to alleviate your fears about my [PR](https://github.com/sabuhish/fastapi-mail/pull/122). And I would be open to that discussion if you are.

**Comment by DiStefano8472** [comment]
Hello together!

@sabuhish 
First of all - great work. I really enjoy the built in jinja support!
But I have the same problem. In our environment we are using Celery for operations which can be done as a job or a scheduled job. But celery is executing tasks and there is no support for awaiting an async result - it is simply not necessary. How much work it would be to integrate a sync version of the send_message function?  For example getting some content or data with httpx client, I can choose between sync or async execution.

@encryptblockr 
Have you already solved your problem?
As a workaround we use `from asgiref.sync import async_to_sync` and a wrapper function to execute send_message. It works - but the code is ugly and I am not sure about negative side effects.

Best regards

Stefan

**Reply by maestro-1** [reply]
Hey @DiStefano8472. So not that much work I imagine. I already made a [PR](https://github.com/sabuhish/fastapi-mail/pull/122) which splits the connection to allow for use of a customizable connection. What this PR does is decentralize the connection engine from the mailing class. This will allow for customizable connections and a few other nice features. Hence you'll be able to use sync libraries like SMTP once this gets merged. And there will also be an entry point for your use case. 

I'm still trying to have a discussion with @sabuhish about the implications of the PR, but if it's approved I believe it would be a great addition. Making fastapi-mail used by any Python framework(sync and async). For now, there is no connection covering sync SMTP objects. You'll have to build that yourself. It should, however, be fairly straightforward as I have included it in the docs and it will get released once/if the PR is merged.
